src/stan_math_backend/Statement_gen.ml

```diff
@@ -6,21 +6,32 @@ open Expression_gen
 let pp_call_str ppf (name, args) = pp_call ppf (name, string, args)
 let pp_block ppf (pp_body, body) = pf ppf "{@;<1 2>@[<v>%a@]@,}" pp_body body
 
+let rec contains_eigen = function
+  | UnsizedType.UArray t -> contains_eigen t
+  | UMatrix | URowVector | UVector -> true
+  | _ -> false
```

Comment:
I would say I'm, "good at this" at least enough to do a review but I think it might be nicer here to just do the 

```suggestion
  | UInt |UReal | UMathLibraryFunction | UFun _ -> false
```

To be explicit about the types you want to check against here